cmd/router/main.go

```diff
@@ -38,8 +39,19 @@ import (
 
 var log = logf.Log.WithName("InferenceGraphRouter")
 
-func callService(serviceUrl string, input []byte) ([]byte, error) {
-	resp, err := http.Post(serviceUrl, "application/json", bytes.NewBuffer(input))
+func callService(serviceUrl string, input []byte, headers http.Header) ([]byte, error) {
+	req, err := http.NewRequest("POST", serviceUrl, bytes.NewBuffer(input))
+	for _, h := range headersToPropagate {
+		_, exists := headers[h]
+		if exists {
+			for _, v := range headers[h] {
+				req.Header.Add(h, v)
+			}
+		}
```

Comment:
this block of code can be simplified to following
```go
if values, ok := headers[h]; ok {
    for _, v := range values {
         req.Header.Add(h, v)
    }
}
```

Reply:
Done